pkgs/by-name/sy/sysdig-cli-scanner/package.nix

```diff
@@ -2,29 +2,42 @@
   stdenv,
   lib,
   fetchurl,
-  makeWrapper,
+  writeShellScript,
 }:
 let
+  pname = "sysdig-cli-scanner";
   versionMetadata = import ./sysdig-cli-scanner.versions.nix;
   fetchForSystem = versionMetadata.${stdenv.system} or (throw "unsupported system ${stdenv.system}");
+
+  wrapper = writeShellScript "${pname}" ''
+    for arg in "$@"; do
+      # We must not pass --dbpath to the cli in case it has been called with --iac
+      # IaC Scanning does not make use of the vulnerability database
+      if [ "$arg" = "--iac" ]; then
+        exec @out@/libexec/${pname} "$@"
+      fi
+    done
+
+    # --dbpath argument is needed for vulnerability scanning mode, otherwise it tries to download
+    # the vulnerability database in the same path as the binary, which is read-only in the case of the
+    # nix store
+    exec @out@/libexec/${pname} \
+      --dbpath="$HOME/.cache/sysdig-cli-scanner/" "$@"
+  '';
 in
 stdenv.mkDerivation {
-  pname = "sysdig-cli-scanner";
+  inherit pname;
```

Comment:
If it's called unwrapped, then the unwrapped `sysdig-cli-scanner` won't be visible, since it's not exposed as a package, but installed into the final derivation.
For the record, this is the construct I followed: 

https://github.com/NixOS/nixpkgs/blob/e74612b0448ba9a726bbc1e98a99be6224ba4565/pkgs/games/fallout-ce/build.nix